factory/glideFactoryLib.py

```diff
@@ -2008,6 +2009,15 @@ def get_submit_environment(
                 full_path_to_proxy = submit_credentials.security_credentials["GlideinProxy"]
                 exe_env.append("GLIDEIN_PROXY_FNAME=%s" % full_path_to_proxy)
 
+                # get the IDTOKEN
+                token_data = ""
+                with open(submit_credentials.identity_credentials["frontend_condortoken"]) as fc:
+                    tok = fc.readlines()
+                    for tln in tok:
+                        token_data = f"{token_data}{tln}"
+
+                exe_env.append("IDTOKEN_DATA=%s" % token_data)
```

Comment:
Adding the token content to the environment there could be the risk of someone looking at it. The Factory should be a secure server, still, the file would be safer. Would be difficult to do something similar to the proxy, using the file name?

Reply:
removed from exe environment visibility.  Required a re-working of credential compression to check if token, proxy, or both provided